test/plugins/compliance_visitor_data_request_plugin_test.rb

```diff
@@ -0,0 +1,159 @@
+require "test_helper"
+
+class ComplianceVisitorDataRequestPluginTest < ActiveSupport::TestCase
+    setup do
+        @data_request_plugin = external_api_clients(:compliance_visitor_data_request_plugin)
+        @visitor_api_namespace = api_namespaces(:compliance_visitor_data_request)
+        @visitor_one = api_resources(:compliance_visitor_data_request_one)
+        @visitor_two = api_resources(:compliance_visitor_data_request_two)
+
+        @namespace_with_email_one = api_namespaces(:namespace_with_email_one)
+        @namespace_with_email_two = api_namespaces(:namespace_with_email_two)
+
+        Sidekiq::Testing.fake!
+
+        # How test fixtures are set up:
+        # By default, "compliance_message_sent" of each visitor api resource is false
+        # By default, each visitor has no submission in an api namespace
+    end
+
+    test "Should not send email to visitors if they already received one" do
+        @visitor_one.properties["compliance_message_sent"] = true
+        @visitor_one.save
+        @visitor_two.properties["compliance_message_sent"] = true
+        @visitor_two.save
+
+        assert_no_difference "ActionMailer::Base.deliveries.size" do
+            perform_enqueued_jobs do
+                @data_request_plugin.run
+                Sidekiq::Worker.drain_all
+            end
+        end
+    end
+
+    test "Should send email if visitor did not receive one and has made a submission" do
```

Comment:
rename test to: `Should send email if compliance_message_sent is set to false on the API resource`